openeo/result.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+from typing import Union, Dict
+
+from openeo.rest import OpenEoClientException
+from openeo.util import ensure_dir
+
+
+class Result:
+    def __init__(self, job):
```

Comment:
I think it would be handy if this `Result` class can also be used for synchronous `executes` (so without a `job` instance). But maybe that is for another iteration.